platforms/windows/SQLitePlugin/SQLitePlugin.cpp

```diff
@@ -0,0 +1,511 @@
+#include "pch.h"
+
+#include "SQLitePlugin.h"
+
+using namespace winrt;
+using namespace winrt::Microsoft::ReactNative;
+using namespace winrt::Windows::Foundation;
+using namespace winrt::Windows::Storage;
+using namespace winrt::Windows::Storage::Streams;
+using namespace winrt::Windows::Security::Cryptography;
+
+namespace SQLitePlugin
+{
+    void SQLitePlugin::Attach(
+        JSValue options,
+        std::function<void(std::string)> onSuccess,
+        std::function<void(std::string)> onFailure) noexcept
+    {
+        SerialReactDispatcher.Post(
+            [onFailure{ std::move(onFailure) }]()
+        {
+            onFailure("attach isn't supported on this platform");
+        });
+    }
+
+    void SQLitePlugin::Close(
+        CloseOptions options,
+        std::function<void(std::string)> onSuccess,
+        std::function<void(std::string)> onFailure) noexcept
+    {
+        SerialReactDispatcher.Post(
+            [
+                options{ std::move(options) },
+                onSuccess{ std::move(onSuccess) },
+                onFailure{ std::move(onFailure) },
+                weak_this = std::weak_ptr(shared_from_this())
```

Comment:
Would this work here?
[wkThis = get_weak()]

Reply:
it's available on winrt components
https://docs.microsoft.com/en-us/windows/uwp/cpp-and-winrt-apis/weak-references

Reply:
Is there a benefit of using winrt implementation over std one for weak pointers?

My understanding is that I would have to inherit the class from 
`winrt::implements<SQLitePlugin, IInspectable>`
instead of 
`std::enable_shared_from_this<SQLitePlugin>`

and then use weak_this.get() instead of weak_this.lock()